components/site/patterns/Create Patterns/CreateServiceModal.js

```diff
@@ -0,0 +1,114 @@
+import * as React from 'react';
+import { useForm, Controller } from 'react-hook-form';
+import { useUID } from '@twilio-paste/core/uid-library';
+import { HelpText } from '@twilio-paste/core/help-text';
+import { Input } from '@twilio-paste/core/input';
+import { Label } from '@twilio-paste/core/label';
+import { Modal, ModalBody, ModalFooter, ModalFooterActions, ModalHeader, ModalHeading } from '@twilio-paste/core/modal';
+import { Box } from '@twilio-paste/core/box';
+import { Button } from '@twilio-paste/core/button';
+import { Stack } from '@twilio-paste/core/stack';
+import PropTypes from 'prop-types';
+
+// These are the default values for the form
+const defaultValues = { serviceName: '', serviceSID: '' };
+
+export const CreateServiceModal = ({ isModalOpen, handleClose, onSubmit }) => {
+  // Form state
+  const { handleSubmit, register, errors, reset, control } = useForm(defaultValues);
+  // Form unique IDs
+  const modalHeadingID = useUID();
+  const serviceNameID = useUID();
+  const serviceSidID = useUID();
+
+  // This resets the form to default values when the modal is closed
+  React.useEffect(() => {
+    if (!isModalOpen) {
+      reset(defaultValues);
+    }
+  }, [isModalOpen, reset]);
+
+  return (
+    <Modal ariaLabelledby={modalHeadingID} isOpen={isModalOpen} onDismiss={handleClose} size="default">
+      <ModalHeader>
+        <ModalHeading as="h3" id={modalHeadingID}>
+          Create new service
+        </ModalHeading>
+      </ModalHeader>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <ModalBody>
+          <Stack orientation="vertical" spacing="space70">
+            <Box>
+              <Label htmlFor={serviceNameID} required>
+                Service Name
+              </Label>
+              {/* This is one of the inputs */}
```

Comment:
I like the comment because it's not obvious that this is an input due to the <Controller> component